tests/multigrid/test_p_multigrid.py

```diff
@@ -26,6 +26,36 @@ def test_reconstruct_degree():
             assert e == PMGPC.reconstruct_degree(elist[0], degree)
 
 
+def test_prolongation_matrix_matfree():
+    from firedrake.preconditioners.pmg import prolongation_matrix_matfree
+
+    tol = 1E-14
+    meshes = [UnitSquareMesh(3, 2, quadrilateral=True)]
+    meshes.append(ExtrudedMesh(meshes[0], layers=2))
+    for mesh in meshes:
```

Comment:
This test is separately testing the quad and hex mesh cases. Can you fixture-ise these so that it is obvious what is going on?

```python
@pytest.fixture(params=["quad", "hex"])
def mesh(request):
    mesh = UnitSquareMesh(3, 2, quadrilateral=True)
    if request.param == "quad":
        return mesh
    elif request.param == "hex":
       return ExtrudedMesh(mesh, layers=2)
    else:
        raise AssertionError


def test_prolongation_matrix_matfree(mesh):
     ndim = mesh.topological_dimension()
     ....
```